src/bin/ion/commands/mod.rs

```diff
@@ -133,3 +136,42 @@ impl WithIonCliArgument for ClapCommand {
         )
     }
 }
+
+trait IoSupport {
+    fn for_each_input(
+        &self,
+        f: impl FnMut(&mut Box<dyn Write>, &str) -> anyhow::Result<()>,
+    ) -> anyhow::Result<()>;
+}
```

Comment:
🗺️ This trait adds an extension method to `ArgMatches` (clap's arguments object). The caller provides a closure that can be run on each `output`/`input` pair. It handles setting up output and coalescing errors.